src/shared/text_or_image.rs

```diff
@@ -54,12 +60,29 @@ live_design! {
 pub struct TextOrImage {
     #[deref] view: View,
     #[rust] status: TextOrImageStatus,
-    // #[rust(TextOrImageStatus::Text)] status: TextOrImageStatus,
     #[rust] size_in_pixels: (usize, usize),
 }
 
 impl Widget for TextOrImage {
     fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
+        // We only handle events if the status is `Image`.
+        if TextOrImageStatus::Image != self.status() { return };
+
+        let image_area = self.view.image(id!(image_view.image)).area();
+
+        match event.hits(cx, image_area) {
+            Hit::FingerDown(_fe) => {
+                cx.set_key_focus(image_area);
+            }
+            Hit::FingerUp(fe) => {
+                if fe.was_tap() {
+                    // Once Clicked, We post an action.
+                    Cx::post_action(TextOrImageAction::ClickImage(self.widget_uid()));
+                }
+            }
```

Comment:
this needs more conditions to be correct:
```suggestion
            Hit::FingerUp(fe) if fe.is_over && fe.is_primary_hit() && fe.was_tap() => {
                Cx::post_action(TextOrImageAction::ClickImage(self.widget_uid()));
            }
```